sdk/src/instruction.rs

```diff
@@ -3,96 +3,124 @@
 use crate::{pubkey::Pubkey, short_vec, system_instruction::SystemError};
 use bincode::serialize;
 use serde::Serialize;
+use thiserror::Error;
 
 /// Reasons the runtime might have rejected an instruction.
-#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+#[derive(Serialize, Deserialize, Debug, Error, PartialEq, Eq, Clone)]
 pub enum InstructionError {
     /// Deprecated! Use CustomError instead!
     /// The program instruction returned an error
+    #[error("generic instruction error")]
     GenericError,
 
     /// The arguments provided to a program instruction where invalid
+    #[error("invalid instruction argument")]
     InvalidArgument,
 
     /// An instruction's data contents was invalid
+    #[error("invalid instruction data")]
     InvalidInstructionData,
 
     /// An account's data contents was invalid
+    #[error("invalid account data for instruction")]
     InvalidAccountData,
 
     /// An account's data was too small
+    #[error("account data too small for instruction")]
     AccountDataTooSmall,
 
     /// An account's balance was too small to complete the instruction
+    #[error("insufficient funds for instruction")]
     InsufficientFunds,
 
     /// The account did not have the expected program id
+    #[error("incorrect program id for instruction")]
     IncorrectProgramId,
 
     /// A signature was required but not found
+    #[error("missing required signature for instruction")]
     MissingRequiredSignature,
 
     /// An initialize instruction was sent to an account that has already been initialized.
+    #[error("already initialized account for instruction")]
     AccountAlreadyInitialized,
 
     /// An attempt to operate on an account that hasn't been initialized.
+    #[error("uninitialized account for instruction")]
     UninitializedAccount,
 
     /// Program's instruction lamport balance does not equal the balance after the instruction
+    #[error("balance mismatch after instruction")]
```

Comment:
This isn't a balance mismatch, its an unbalanced lamport balance after the instruction

Reply:
hehe, "unbalanced balance", gotta be a better way to say that.  Also, @t-nelson, please avoid the word "lamport" in user-facing messages. It's all SOL for them. Oftentimes best to say "amount" or "balance" - only use SOL if the added precision is helpful.

Reply:
"unbalanced balance" is _exactly_ what I was dancing around :laughing:

Reply:
"Total account balances before and after instruction don't match"?